---
layout: post
location: Master's thesis research project
title: Design of a machine learning approach for bacterial microorganisms detection in metagenomic sequencing without alignment procedures
image: /assets/img/caribou-algo.png
category: genomics
tag: metagenomics
description: Data from metagenomic sequencing experiments are complex to analyse because of their size, knowledge contained in databases and performances of already available tools, based on multiple criteria. The most precise methods are are based on alignment procedures, but usually require large computationnal resources. Therefore, this research project aims to design and implement a method for classifying metagenomic sequencing data at a large scale using alignment-free methods. The research work is conceptual in nature but it was also implemented to test it's performances on artificial data. The main part of the project is concentrated on large scale data handling and machine learning. To permit training models on whole genomes and subsequent alignment-free classification of metagenomic sequencing data, the k-mers method for profiles extractions was used. Classification is done in a reccurent manner using a "top-down" strategy. Therefore, a binary classification permits to extract the bacterial sequences then the multiclass classification is done on multiple taxonomic levels for the sequences that were identified at a previous level, going from the largest level to the most precise one. For each step two classical algorithms and three neural networks are trained on whole genomes from the  "Genome Taxonomy Database" and tested using cross validation on reads simulated from training genomes.
contributors: Amine M. Remita, Abdoulaye Banir√© Diallo (PI)
tasks: Design, Implementation, Testing
tools: Python 3, Tensorflow, Keras, Scikit-Learn, Ray, Pandas, Numpy, PyArrow, Biopython, <a href="https://github.com/HadrienG/InSilicoSeq" target="_blank">InSilicoSeq</a>
article: 
github: https://github.com/bioinfoUQAM/Caribou
website: 
---