---
date: 2019-09-04
---

# Inonotus Obliquus article

## RNA-Seq de Novo Assembly and Differential Transcriptome Analysis of Chaga (*Inonotus obliquus*) Cultured with Different Betulin Sources and the Regulation of Genes Involved in Terpenoid Biosynthesis

:::{margin}
### Team:
* Dre Narimene Fradj
* Dre Natacha MÃ©rindol
* Dre Fatima Awwad
* Pr Yacine Boumghar
* Pr Hugo Germain
* Pre Isabel DesgagnÃ©-Penix
:::

### ğŸ¯ Project objective
Internship project at the end of the Bachelor's degree in Biological & Ecological Sciences.

This project aimed to learn how to work with sequencing data and to help Dre Narimene Fradj in completing her PhD project.

### ğŸ“‹ Project description
Working with RNA-Seq sequencing data, the project aimed to characterise the metabolic pathway leading to terpenoids molecules (eg., betulin, betulinic acid and inotodiol) production in *Inonotus Obliquus* (Chaga fungi).

Chaga was cultured in three environments to determine the optimal environment for terpenoids production. From these three cultures, RNA was extracted and sequenced to study their transcriptome.

To do so, reads were aligned to a consensus genome and were then assigned functions using the Gene Ontology database. This research project was part of Dre Narimene Fradj's PhD project and led to an article in IJMS.

### ğŸ¨ Implementation decisions
* Statistical analysis and graphs were performed with R programming language.
* Usage of Gene Ontologies for mapping reads

### ğŸ§¾ Key takeaway
* First contact in working with sequencing data
* Reinforcement of statistical analysis and use of R programming language

::::{grid} 1 1 2 2

:::{card}

### ğŸ‘¨â€ğŸ’» Contribution:
* RNA-seq data analysis between three culture mediums
* Data vizualisation and exploration
* Graphics and layouts
:::

:::{card}

### ğŸ›  Tools:
* R programming language
* ggplot2
* tidyr
* plotly graphs
:::

::::

{button}`Article <https://doi.org/10.3390/ijms20184334>`
{button}`Github repository <https://github.com/nicdemon/inonotus-obliquus.git>`